SourceCode/GPS/Forms/Controls.Designer.cs

```diff
@@ -1753,25 +1753,52 @@ private void btnChangeMappingColor_Click(object sender, EventArgs e)
         }
         private void btnYouSkipEnable_Click(object sender, EventArgs e)
         {
-            yt.alternateSkips = !yt.alternateSkips;
-            if (yt.alternateSkips)
-            {
-                btnYouSkipEnable.Image = Resources.YouSkipOn;
-                //make sure at least 1
-                if (yt.rowSkipsWidth < 2)
-                {
-                    yt.rowSkipsWidth = 2;
-                    cboxpRowWidth.Text = "1";
-                }
-                yt.Set_Alternate_skips();
-                yt.ResetCreatedYouTurn();
-                if (!yt.isYouTurnBtnOn) btnAutoYouTurn.PerformClick();
-            }
-            else
+
+            yt.skip_mode++;
+            if ((int)yt.skip_mode > 2) yt.skip_mode = SkipMode.Normal;
```

Comment:
Can we do this with a switch statement that explicitly converts from one mode to the next?

```suggestion
            switch (yt.skip_mode)
            {
                case SkipMode.Normal:
                    yt.skip_mode = SkipMode.Alternative;
                    break;
                case SkipMode.Alternative:
                    yt.skip_mode = SkipMode.ignoreWorkedTracks;
                    break;
                case SkipMode.ignoreWorkedTracks:
                    yt.skip_mode = SkipMode.Normal;
                    break;
            }
```

Even though this is more code, I think this would be more readable. (Maybe move it into a separate method `NextSkipMode`, that way it would become even more understandable.)

Reply:
Thanks - i'll adjust and create a new suggestion